amplipi/updater/asgi.py

```diff
@@ -59,10 +62,42 @@ class ReleaseInfo(BaseModel):
 dir_path = os.path.dirname(real_path)
 app.mount("/static", StaticFiles(directory=f"{dir_path}/static"), name="static")
 
-HOME = f"{os.environ.get('HOME')}/amplipi-dev" # standard install directory
+INSTALL_DIR = os.getenv('INSTALL_DIR', os.getcwd())
+
+# if we have a broken configuration, the updater should still function
+# as a failsafe
+try:
+  with open(f"{INSTALL_DIR}/house.json") as f:
+    config = json.load(f)
+except:
+  config = {"info": {}}
+
+# The `auto_error` arg below prevents the HTTPBasic middleware from prompting for a password
+# when initial loading of a page while an admin hash is not configured.
+security = HTTPBasic(auto_error = True if 'admin_password_hash' in config['info'].keys() else False)
```

Comment:
between this and lines 84-86, we ought not prompt nor require authorization unless an `admin_password_hash` is set. Without the initialization of `HTTPBasic` this way, configurations without a password set would always prompt... and `check_password` below would accept anything :upside_down_face: